vm/src/builtins/set.rs

```diff
@@ -473,12 +473,18 @@ impl PySet {
 
     #[pymethod(magic)]
     fn repr(zelf: PyRef<Self>, vm: &VirtualMachine) -> PyResult {
+        let class_name = zelf.class().name();
         let s = if zelf.inner.len() == 0 {
-            "set()".to_owned()
+            format!("{}()", class_name)
         } else if let Some(_guard) = ReprGuard::enter(vm, zelf.as_object()) {
-            zelf.inner.repr(vm)?
+            let inner_repr = zelf.inner.repr(vm)?;
```

Comment:
inner_repr can be very long. in that case, using it as argument for another formatting will cost a lot. Could you change `PySetInner::repr` to take class name like below?

```rust
fn repr(&self, class_name: Option<&str>, vm: &VirtualMachine) -> PyResult<String>
```

This approach can be used for other reprs too.